patches/server/0969-Added-config-option-to-hide-Loottables-and-PDC.patch

```diff
@@ -0,0 +1,46 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jayson <jayson2001@hotmail.be>
+Date: Thu, 23 Mar 2023 18:20:19 +0100
+Subject: [PATCH] Added config option to hide Loottables and PDC
+
+
+diff --git a/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java b/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java
+index 51cf0014c4229fc8671804d885b6381996810130..7b0f4be4983fa66bbfda436ab4f9f058f971f0e8 100644
+--- a/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java
++++ b/src/main/java/io/papermc/paper/configuration/WorldConfiguration.java
+
+@@ -78,6 +79,13 @@ public class WorldConfiguration extends ConfigurationPart {
+                 public boolean hideDurability = false;
+                 public boolean hideItemmetaWithVisualEffects = false;
+             }
++
++            public EntityData entitydata;
++
++            public class EntityData extends ConfigurationPart {
++                public boolean hideLootTables = false;
++                public boolean hidePDC = true;
++            }
+         }
+ 
+         public AntiXray antiXray;
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java
+index 1b248db497500aa6bd346b306dcb908af77626f3..ea6ff14fea3f658938d6c6189eb3ee67eaa7c560 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java
+ 
+@@ -258,7 +262,14 @@ public abstract class BlockEntity {
+     // CraftBukkit end
+     // Paper start
+     public CompoundTag sanitizeSentNbt(CompoundTag tag) {
+-        tag.remove("PublicBukkitValues");
++
++        if(level.paperConfig().anticheat.obfuscation.entitydata.hideLootTables && tag.contains("LootTable")){
++            tag.remove("LootTable");
++            tag.remove("LootTableSeed");
++        };
++        if(level.paperConfig().anticheat.obfuscation.entitydata.hidePDC && tag.contains("PublicBukkitValues") ){
++            tag.remove("PublicBukkitValues");
++        };
```

Comment:
Do you need the extra contains() checks? remove should still succeed if it doesn't have that tag key. It's just one less lookup

Reply:
I just added this just to be sure. I will test it with the rest.